regexec.c

```diff
@@ -8804,22 +8794,6 @@ NULL
         case WHILEM_B_min_fail: /* just failed to match B in a minimal match */
             cur_curlyx = ST.save_curlyx;
 
-            if (cur_curlyx->u.curlyx.count >= /*max*/ARG2(cur_curlyx->u.curlyx.me)) {
-                /* Maximum greed exceeded */
-                if (cur_curlyx->u.curlyx.count >= REG_INFTY
-                    && ckWARN(WARN_REGEXP)
-                    && !reginfo->warned)
-                {
-                    reginfo->warned	= TRUE;
-                    Perl_warner(aTHX_ packWARN(WARN_REGEXP),
-                        "Complex regular subexpression recursion "
-                        "limit (%d) exceeded",
-                        REG_INFTY - 1);
-                }
-                cur_curlyx->u.curlyx.count--;
-                CACHEsayNO;
-            }
```

Comment:
I think this removes too much: the outer block looks to me like it is checking whether we've reached the max allowed for this (minimal) repetition. I can have a dig to verify, but I expect @demerphq will know.

If I'm correct, then we should really have a test that fails when you remove this. (I'm a bit surprised we don't have any tests of the warning, for that matter.)

Reply:
Thank you, I believe you are correct.

Reply:
@khwilliamson I think this got merged without fixing the above over-removal, please could you verify?